Documentation/kubernetes.md

```diff
@@ -21,28 +21,41 @@ You will need to use the yaml files from the [1.5 folder](/cluster/examples/kube
 
 ### Prerequisites
 
-To make sure you have a Kubernetes cluster that is ready for `Rook`, you can [follow these quick instructions](k8s-pre-reqs.md), including:
-- The `kubelet` requires access to `modprobe` and `rbd` on host
+To make sure you have a Kubernetes cluster that is ready for `Rook`, you can [follow these quick instructions](k8s-pre-reqs.md).
 
 Note that we are striving for even more smooth integration with Kubernetes in the future such that `Rook` will work out of the box with any Kubernetes cluster.
 
 If you are using `dataDirHostPath` to persist rook data on kubernetes hosts, make sure your host has at least 5GB of space available on the specified path.
 
 ### Deploy Rook
 
-With your Kubernetes cluster running, Rook can be setup and deployed by simply creating the rook-operator deployment and creating a rook cluster. To customize the operator settings, see the [Operator Helm Chart](helm-operator.md).
+With your Kubernetes cluster running, Rook can be setup and deployed by simply creating the rook-operator deployment and creating a rook cluster. The rook-operator will automatically deploy a Rook-agent on all the nodes. Rook-agents perform operations to consume storage from the cluster. Refer to the [Rook-agent design document](/design/local-node-agent.md) for more information.
+
+To customize the operator settings, see the [Operator Helm Chart](helm-operator.md).
 
 ```bash
 cd cluster/examples/kubernetes
 kubectl create -f rook-operator.yaml
 
-# verify the rook-operator pod is in the `Running` state before proceeding
-kubectl get pod
+# verify the rook-operator and rook-agents pods are in the `Running` state before proceeding
+kubectl -n rook-system get pod
 ```
 
+---
+**Restart Kubelet (K8S 1.7 and older)**
+
+After the Rook operator and Rook agents have been deployed, The Kubelet process on all nodes will require a restart. As part of their initial setup, the Rook agents deploy and configure a Flexvolume plugin in order to integrate with Kubernetes' volume controller framework. Kubernetes v1.8+, the [dynamic Flexvolume plugin discovery](https://github.com/kubernetes/community/pull/833) will find and initialize our plugin, but in older versions of Kubernetes a manual restart of the Kubelet will be required. 
```

Comment:
maybe start this sentence with "For versions of Kubernetes prior to 1.8, the Kubelet process on all nodes will require a restart after the Rook operator and Rook agents have been deployed." just to reinforce that this is not needed in 1.8.

I also don't think we need to get into the details of the plugin discovery pull request, this is supposed to be a quick and easy flow for getting Rook running.

Reply:
By specifying the issue, I just wanted to emphasize that this issue is due to a kubernetes limitation. Just in case it wasnt clear.